### PR TITLE
#165305224 Implement social login

### PR DESCRIPTION
**Pull request by jkamz**
### What does this PR do?
Implements social login

### Description of Task(s) to be completed?
- implement functionality to get access token from facebook/ google
- include social media icons on both signup and sign in pages
- send login request to ah backend using access token and provider details
- show spinner when logging in
- disable buttons after response
- update failing tests
- add tests for social login

### How should this be manually tested?

Clone this repository on local machine and CD into project folder
`$ git clone git@github.com:andela/ah-the-jedi-frontend.git`
`$ cd ah-the-jedi-frontend`

Check out to the PR's branch
`$ git checkout ft-social-login-165305224`

install dependencies and run the dev server
`npm install`
`npm run start:dev`

navigate to login and signup page and interact with social login buttons

### What are the relevant PT stories?
[#165305224](https://www.pivotaltracker.com/story/show/165305224)

### Screenshots
image

image

### CODE REVIEW

**Review by Benkimeric — APPROVED** [review]
Nice implementation on social login feature. I like the design and the integration with existing features such as disabling existing login buttons while loading.

**Review by Siffersari — APPROVED** [review]
Good job changing the Icons for the Social Logins to match the color code and to be more specific. I love the spacing and the design. Good work with the disabled buttons when the request is made and adding the Loader to the button. 

Commendable use of self documenting code

**Review by AlvinMugambi — APPROVED** [review]

File: src/components/auth/Social.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
+import { GoogleLogin } from 'react-google-login';
```
Comment (katherine95) [inline src/components/auth/Social.js:5]: Great job in making use of react social login libraries. i.e FacebookLogin and GoogleLogin.

**Review by katherine95 — APPROVED** [review]
Good job on writing tests. Otherwise looks good to me.

**Pull request merged**